content/articles/spring-webflux/index.md

```diff
@@ -0,0 +1,315 @@
+
+In this tutorial, we are going to create a student management system using Spring webflux and MongoDB.
+
+
+### Prerequisites
+- [JDK](https://www.oracle.com/java/technologies/javase-downloads.html) installed on your computer.
+- Your favorite IDE or editor installed.
+- Knowledge of [Java](https://www.tutorialspoint.com/java/index.htm) and [Spring Boot](https://spring.io/projects/spring-boot).
+- Knowledge of [Mongodb](https://www.mongodb.com/).
+
+### Streams API
+Software developers at Netflix, Twitter, Pivotal, and Redhat created the streams API. Streams API defines four interfaces below.
+
+**Publisher**
+The publisher interface emits events to subscribers based on the request sent by the subscribers. Thus, a single publisher can serve several subscribers.
+
+```java
+public interface Publisher<T> 
+{
+    public void subscribe(Subscriber<? super T> s);
+}
+```
+
+**Subscriber**
+`Subscriber` interface listens and receives events from the `Publisher` interface. `Subscriber` interface has four methods to handle the response from the `Publisher` interface.
+
+```java
+public interface Subscriber<T> 
+{
+    public void onSubscribe(Subscription s);
+    public void onNext(T t);
+    public void onError(Throwable t);
+    public void onComplete();
+}
+```
+
+**Subscription**
+`Subscription interface defines a one-to-one relationship between the `Publisher` and `Subscriber` interfaces. It can be used to request data and also cancel the request.
+
+```java
+public interface Subscription<T> 
+{
+    public void request(long n);
+    public void cancel();
+}
+```
+
+**Processor**
+`Processor` interface represents the processing stage containing both the `Publisher` and the `Subscriber`.
+
+```java
+public interface Processor<T, R> extends Subscriber<T>, Publisher<R> 
+{
+
+}
+```
+
+> There are two popular implementations of the reactive streams, [RxJava](https://github.com/ReactiveX/RxJava) and [Project reactor](https://projectreactor.io/).
+
+### Spring webflux
+Spring webflux is simmilar to Spring MVC , but it suppoerts reactive and non blocking streams.
+
+Webflux has two publishers:
+
+**Mono**
+`Mono` publisher that returns 0 or 1 element.
+```java
+Mono<String> mono = Mono.just("Jonh");
+Mono<String> mono = Mono.empty();
+```
+
+**Flux**
+`Flux` is a publisher that emits 0 or N elements.
+
+```java
+Flux<String> flux = Flux.just("x", "y", "z");
+Flux<String> flux = Flux.fromArray(new String[]{"x", "y", "z"});
+Flux<String> flux = Flux.fromIterable(Arrays.asList("x", "y", "z"));
+ 
+//To subscribe, call the method
+ 
+flux.subscribe();
+```
+
+### Application Setup
+We are going to use [spring initialr](https://start.spring.io/) to generate our application startup code.
```

Comment:
Spelling error. Change *initialr* to *initializer*. Also, before every instance of the common noun *spring initializer* add the word *the*. [Common nouns](https://www.grammarly.com/blog/common-noun/) need to have words like *a*, *the*, etc. in front of them.